### PR TITLE
UefiCpuPkg: Additional CodeQL fixes

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Includes fixes for an additional set of CodeQL queries.

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Build UefiCpuPkg and boot changes on QemuQ35Pkg to EFI shell.

## Integration Instructions

N/A

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

File: UefiCpuPkg/Library/RegisterCpuFeaturesLib/CpuFeaturesInitialize.c [hunk 1]
```diff
@@ -130,7 +130,13 @@ CpuInitDataInitialize (
   GetNumberOfProcessor (&NumberOfCpus, &NumberOfEnabledProcessors);
 
   CpuFeaturesData->InitOrder = AllocatePages (EFI_SIZE_TO_PAGES (sizeof (CPU_FEATURES_INIT_ORDER) * NumberOfCpus));
-  ASSERT (CpuFeaturesData->InitOrder != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
```
Comment (makubacki) [inline UefiCpuPkg/Library/RegisterCpuFeaturesLib/CpuFeaturesInitialize.c:151]: Looking at this again, there's a lot of allocations in the function. I'm planning to:

1. Initialize all of the pointers that get assigned memory to NULL (at the beginning of the function)
2. Route all cases where an allocation fails to a label at the end of the function that frees buffers and restores external pointers (like `CpuFeaturesData->InitOrder`) to a known value (e.g. `NULL`) on an allocation failure.

Comment (makubacki) [reply UefiCpuPkg/Library/RegisterCpuFeaturesLib/CpuFeaturesInitialize.c:151]: Updated

**Review by apop5 — APPROVED** [review]

File: UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c [hunk 2]
```diff
@@ -432,7 +467,16 @@ SmmInitPageTable (
     // Add pages to page pool
     //
     FreePage = (LIST_ENTRY *)AllocatePageTableMemory (PAGE_TABLE_PAGES);
-    ASSERT (FreePage != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (FreePage == NULL) {
+      ASSERT (FreePage != NULL);
```
Comment (kuqin12) [inline UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c]: Should we continue here?

**Comment by makubacki** [comment]
Note that I split out the filters into their own PR since they're more straightforward and needed to get cleaner results in many other packages (that use UefiDevicePathLib).

https://github.com/microsoft/mu_basecore/pull/264

**Review by kuqin12 — APPROVED** [review]

File: UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c (hunk 2 above)
Comment (makubacki) [reply UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c]: That seems to be a good idea. Can you confirm the latest code matches what you were thinking?

Comment (kuqin12) [reply UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c]: Yep, thanks!

**Pull request merged**